storage/Devices/MsdosImpl.cc

```diff
@@ -63,6 +63,29 @@ namespace storage
 
 
     void
+    Msdos::Impl::delete_partition(Partition* partition)
+    {
+	PartitionType old_type = partition->get_type();
+	unsigned int old_number = partition->get_number();
+
+	PartitionTable::Impl::delete_partition(partition);
+
+	// After deleting a logical partition the numbers of partitions with
+	// higher numbers are shifted.
+	if (old_type == PartitionType::LOGICAL)
+	{
+	    vector<Partition*> partitions = get_partitions();
+            for (Partition* tmp : partitions)
+	    {
+		unsigned int number = tmp->get_number();
+		if (number > old_number)
```

Comment:
I know that in this context this cannot happen, but I'd feel a lot more comfortable with `if (number != old_number)`. Just to make sure.